### PR TITLE
Update: Use remark for parsing (fixes #38)

### LINKED ISSUE #38: Consider using Remark for parsing
Remark is a dedicated library for parsing Markdown files to an AST. It would probably be better to use a dedicated parser instead of rolling out our own.

### PR DESCRIPTION
**Pull request by btmills**
/cc @BYK

### CODE REVIEW

File: lib/processor.js [hunk 1]
```diff
@@ -9,179 +9,51 @@
...
-        }
-
-        captures = match(/^( {0,3})(`{3,}|~{3,})\h*(?:js|javascript|node|jsx)\h*\r?\n/i);
-        if (captures) {
-            return {
-                indent: captures[1],
-                fence: captures[2]
-            };
-        }
+function traverse(node, callbacks) {
```
Comment (BYK) [inline lib/processor.js]: Hah, may be we should start a project called `marktraverse` for this stuff :)

File: lib/processor.js [hunk 2]
```diff
@@ -9,179 +9,51 @@
...
  */
 function preprocess(text) {
-    blocks = parse(text);
+    var ast = remark.parse(text);
+
+    blocks = [];
+    traverse(ast, {
+        "code": function(node) {
+            if (node.lang && SUPPORTED_SYNTAXES.indexOf(node.lang.toLowerCase()) >= 0) {
+                blocks.push(node);
```
Comment (BYK) [inline lib/processor.js]: Why not `blocks.push(node.value)` directly if you're not using any other property of the node afterwards?

**Comment by BYK** [comment]
Nice. I think the build failure is not directly related.

File: lib/processor.js (hunk 1 above)
Comment (btmills) [inline lib/processor.js]: I thought about shaving that yak just for fun, but it was late, and I didn't want to write all the tests I'd have to write for a new module :)

**Comment by btmills** [comment]
> I think the build failure is not directly related.

Yeah, that had to do with `eslint-config-eslint` and `eslint@2.0.0-beta.1` as a `peerDependency`. There's another PR that attempted to solve that, but I'll have to try something else today.

**Comment by btmills** [comment]
Build failures were solved in #41.

**Pull request merged**